akka-samples/akka-sample-cluster-java/src/main/java/sample/cluster/factorial/FactorialFrontendMain.java

```diff
@@ -30,10 +34,22 @@ public void run() {
 
     //#registerOnRemoved
     Cluster.get(system).registerOnMemberRemoved(new Runnable() {
-        @Override
-        public void run() {
-            system.terminate();
-        }
+      @Override
+      public void run() {
+        // exit JVM when ActorSystem has been terminated
+        final Runnable exit = new Runnable() {
+          @Override
+          public void run() {
+            System.exit(-1);
+          }
+        };
+        system.registerOnTermination(exit);
+        // if the ActorSystem does not terminate
```

Comment:
As mentioned elsewhere, I'd reword to  "in case ActorSystem shutdown takes longer than 10 seconds, exit the JVM forcefully anyway"?